lib/saml_idp/request.rb

```diff
@@ -154,6 +154,10 @@ def valid?
       return true
     end
 
+    def signed?
+      document.signed? || !!document.send(:options_have_signature, self.options)
```

Comment:
Is it worth considering to make the method public so we don't need to `.send` ?

https://github.com/18F/saml_idp/blob/9a219c3c0e1f379ff315bc9ccd9e15f573d35f52/lib/saml_idp.rb#L75

Reply:
Or maybe even reimplement the logic, since it's not particularly complex.

```rb
!!self.options[:get_params]&.key?(:Signature)
```

Reply:
yeah that's a good point - I don't normally like code duplication but this gem is such a mess that I don't think it hurts much. Thanks!